### PR TITLE
Passive templates with runtime error 

### PR DESCRIPTION
**Pull request by xm1k3**
## Proposed changes

The Problem was on:

```go
...
        if options.Options.OfflineHTTP {
		template.compileOfflineHTTPRequest(options)
		return nil
	}
...
```
`template.Executer` was always nil

## Checklist

- [x] Pull request is created against the [dev](https://github.com/projectdiscovery/nuclei/tree/dev) branch
- [ ] All checks passed (lint, unit/integration/regression tests etc.) with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Review by Mzack9999 — CHANGES_REQUESTED** [review]
Offline matching is done only with those templates that have GET request to root URL, we should just return an error as follows:
```go
func (template *Template) compileOfflineHTTPRequest(options protocols.ExecuterOptions) error {
...
	if len(operatorsList) > 0 {
		options.Operators = operatorsList
		template.Executer = executer.NewExecuter([]protocols.Request{&offlinehttp.Request{}}, &options)
		return nil
	}
	return errors.New("template can't be used for offline matching")
}
```

File: v2/pkg/catalog/loader/loader.go [hunk 1]
```diff
@@ -251,10 +251,10 @@ func areWorkflowTemplatesValid(store *Store, workflows []*workflows.WorkflowTemp
 }
 
 func isParsingError(message string, template string, err error) bool {
-	if err == templates.ErrCreateTemplateExecutor {
+	if err == filter.ErrExcluded {
```
Comment (Mzack9999) [inline v2/pkg/catalog/loader/loader.go:254]: errors.Is(err, filter.ErrExcluded)

File: v2/pkg/catalog/loader/loader.go [hunk 2]
```diff
@@ -251,10 +251,10 @@ func areWorkflowTemplatesValid(store *Store, workflows []*workflows.WorkflowTemp
...
 		return false
 	}
-	if err == filter.ErrExcluded {
+	if err == templates.ErrCreateTemplateExecutor {
```
Comment (Mzack9999) [inline v2/pkg/catalog/loader/loader.go:257]: same here

File: v2/pkg/templates/compile.go [hunk 3]
```diff
@@ -142,13 +142,16 @@ func (template *Template) compileProtocolRequests(options protocols.ExecuterOpti
 		return fmt.Errorf("no requests defined for %s", template.ID)
 	}
 
+	var requests []protocols.Request
+
 	if options.Options.OfflineHTTP {
-		template.compileOfflineHTTPRequest(options)
+		err := template.compileOfflineHTTPRequest(options)
```
Comment (Mzack9999) [inline v2/pkg/templates/compile.go:148]: I think it's equal to just:
```go
return template.compileOfflineHTTPRequest(options)
```

**Review by Mzack9999 — CHANGES_REQUESTED** [review]

File: v2/pkg/templates/compile.go [hunk 4]
```diff
@@ -219,5 +218,8 @@ mainLoop:
 	if len(operatorsList) > 0 {
 		options.Operators = operatorsList
 		template.Executer = executer.NewExecuter([]protocols.Request{&offlinehttp.Request{}}, &options)
+		return nil
 	}
+	return errors.New("template can't be used for offline matching")
+
```
Comment (Mzack9999) [inline v2/pkg/templates/compile.go:224]: extra new line

**Review by Mzack9999 — CHANGES_REQUESTED** [review]

**Review by Mzack9999 — APPROVED** [review]
lgtm - error message has been made more explicit

**Review by ehsandeep — CHANGES_REQUESTED** [review]
@xm1k3 @Mzack9999 passive template only works with file input containing http response. Can we also do option/input validation before template parsing for input requirements?

For example -
```console
./nuclei -passive

                     __     _
   ____  __  _______/ /__  (_)
  / __ \/ / / / ___/ / _ \/ /
 / / / / /_/ / /__/ /  __/ /
/_/ /_/\__,_/\___/_/\___/_/   v2.8.4-dev

		projectdiscovery.io

[INF] Using Nuclei Engine 2.8.4-dev (development)
[INF] Using Nuclei Templates 9.3.2 (latest)
[INF] Templates added in last update: 57
[INF] Templates loaded for scan: 655
[INF] Targets loaded for scan: 0
[FTL] Could not run nuclei: missing required input (http response) to run passive templates
```

**Comment by xm1k3** [comment]
@ehsandeep added error message on input validation

`nuclei --passive`

```
...
...
[INF] Targets loaded for scan: 0
[INF] No results found. Better luck next time!
[FTL] Could not run nuclei: missing required input (http response) to run passive templates
```

**Comment by Mzack9999** [comment]
Due to GH inconsistent behavior - let's close this PR and create a new one

**Pull request closed without merging**